utils/check_modular_conversion.py

```diff
@@ -135,14 +141,32 @@ def guaranteed_no_diff(modular_file_path, dependencies, models_in_diff):
     skipped_models = set()
     non_matching_files = 0
     ordered_files, dependencies = find_priority_list(args.files)
-    for modular_file_path in ordered_files:
-        is_guaranteed_no_diff = guaranteed_no_diff(modular_file_path, dependencies, models_in_diff)
-        if is_guaranteed_no_diff:
-            model_name = modular_file_path.rsplit("modular_", 1)[1].replace(".py", "")
-            skipped_models.add(model_name)
-            continue
-        non_matching_files += compare_files(modular_file_path, args.fix_and_overwrite)
-        models_in_diff = get_models_in_diff()  # When overwriting, the diff changes
+    if args.fix_and_overwrite or args.num_workers == 1:
+        for modular_file_path in ordered_files:
+            is_guaranteed_no_diff = guaranteed_no_diff(modular_file_path, dependencies, models_in_diff)
+            if is_guaranteed_no_diff:
+                model_name = modular_file_path.rsplit("modular_", 1)[1].replace(".py", "")
+                skipped_models.add(model_name)
+                continue
+            non_matching_files += compare_files(modular_file_path, args.fix_and_overwrite)
+            models_in_diff = get_models_in_diff()  # When overwriting, the diff changes
+    else:
+        new_ordered_files = []
+        for modular_file_path in ordered_files:
+            is_guaranteed_no_diff = guaranteed_no_diff(modular_file_path, dependencies, models_in_diff)
+            if is_guaranteed_no_diff:
+                model_name = modular_file_path.rsplit("modular_", 1)[1].replace(".py", "")
+                skipped_models.add(model_name)
+            else:
+                new_ordered_files.append(modular_file_path)
+
+        new_ordered_files = ordered_files
+        import multiprocessing
+
+        with multiprocessing.Pool(4) as p:
+            outputs = p.map(compare_files, new_ordered_files)
+        for output in outputs:
+            non_matching_files += output
```

Comment:
Just curious couldn't we get unexpected results with multiprocessing here, for example

new_ordered_files = [modular_B(A), modular_C(B)]

With multiprocessing both of them would be converted "at the same time", but should be converted sequentially because C depends on the converted B model.

Reply:
`new_ordered_files`  is computed sequentially. And the MP is only used in the case where `fix_and_overwrite=False`, so there is no file change in this case :-)